### PR TITLE
Major/uppsf 1392 migrate logger

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**

### CODE REVIEW

File: transport/logging_roundtrip.go [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
+
+	"github.com/Financial-Times/go-logger/v2"
+	transactionidutils "github.com/Financial-Times/transactionid-utils-go"
 )
 
 type loggingRoundTripper struct {
-	L  *logrus.Logger
-	Rt http.RoundTripper
+	log *logger.UPPLogger
+	Rt  http.RoundTripper
```
Comment (karsov) [inline transport/logging_roundtrip.go:13]: As with the upper, the struct is not public so no need for public property.
Also a longer name may be better to read.

File: transport/logging_roundtrip.go [hunk 2]
```diff
@@ -38,7 +34,7 @@ func (lrt *loggingRoundTripper) RoundTrip(req *http.Request) (*http.Response, er
 	response, err := lrt.Rt.RoundTrip(req)
 	elapsed := time.Since(t)
 
-	withFields := lrt.L.WithFields(logrus.Fields{
+	withFields := lrt.log.WithFields(map[string]interface{}{
```
Comment (karsov) [inline transport/logging_roundtrip.go:37]: Let's log the full URL (host + path) in requestUri. Logging only the path makes debugging very confusing.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module github.com/Financial-Times/go-ft-http
+
+go 1.14
+
+require (
+	github.com/Financial-Times/go-logger/v2 v2.0.1
+	github.com/Financial-Times/service-status-go v0.0.0-20160323111542-3f5199736a3d
+	github.com/Financial-Times/transactionid-utils-go v0.2.0
+	github.com/hashicorp/go-version v1.2.0 // indirect
+	github.com/stretchr/testify v1.2.1
```
Comment (karsov) [inline go.mod:10]: We can `go get -u github.com/stretchr/testify`

File: transport/logging_roundtrip.go (hunk 2 above)
Comment (ivan-p-nikolov) [reply transport/logging_roundtrip.go:37]: It was meant to copy the log in http-handlers-go.

Comment (ivan-p-nikolov) [reply transport/logging_roundtrip.go:37]: Added a separate field with the full requestURL in it

File: transport/transport.go [hunk 4]
```diff
@@ -38,8 +37,8 @@ func NewTransport() *DelegatingTransport {
 }
 
 // NewTransport returns a delegating transport which uses the http.DefaultTransport
-func NewLoggingTransport(logger *logrus.Logger) *DelegatingTransport {
-	return (&DelegatingTransport{delegate: &loggingRoundTripper{L: logger, Rt: http.DefaultTransport}}).WithTransactionIDFromContext()
+func NewLoggingTransport(logger *logger.UPPLogger) *DelegatingTransport {
+	return (&DelegatingTransport{delegate: &loggingRoundTripper{log: logger, tripper: http.DefaultTransport}}).WithTransactionIDFromContext()
```
Comment (karsov) [inline transport/transport.go:41]: Could we just make this very long line more readable?

**Comment by karsov** [comment]
I'd suggest that we use "v1.0.0" for this because it has been working for a long time now.
... and leave the redesign to v2?

**Comment by ivan-p-nikolov** [comment]
I've opened an new ticket for the refactoring here https://financialtimes.atlassian.net/browse/UPPSF-1410

**Pull request merged**